interfaces/apparmor/backend.go

```diff
@@ -300,12 +300,12 @@ func profileGlobs(snapName string) []string {
 // - snap.core.NNNN.usr.lib.snapd.snap-confine
 // - var.lib.snapd.snap.core.NNNN.usr.lib.snapd.snap-confine
 // - snap-confine.core.NNNN
-// TODO: also the "snapd" snap here soon
+// - snap-confine.snapd.NNNN
 func profileIsRemovableOnCoreSetup(fn string) bool {
 	bn := path.Base(fn)
 	if strings.HasPrefix(bn, ".") {
 		return false
-	} else if strings.HasPrefix(bn, "snap") && !strings.HasPrefix(bn, "snap-confine.core.") && !strings.Contains(bn, "usr.lib.snapd.snap-confine") {
+	} else if strings.HasPrefix(bn, "snap") && !strings.HasPrefix(bn, "snap-confine.core.") && !strings.HasPrefix(bn, "snap-confine.snapd.") && !strings.Contains(bn, "usr.lib.snapd.snap-confine") {
```

Comment:
+1 but the length of this line is ridiculous now. Would you mind taming that?

Reply:
All this LGTM. I don't mind the length as it's clear (feel free to add some newlines if go fmt allows). Remember we all bikeshedded on the calculation a bit and landed on this. ;)